### PR TITLE
DOC: update joblib installation procedure in the documentation

### LINKED ISSUE #339: Update installation instructions in docs
Documentation currently recommends `easy_install` at the first option, which is not really the recommended way to do things these days. Probably would be better to recommend `pip install` and also mention that it can be installed with common scientific python packages like anaconda using `conda install`.

### PR DESCRIPTION
**Pull request by aabadie**
Update installation procedure in the documentation. I think everyone now uses `pip` or `conda`. Fix #339.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -15,11 +15,15 @@ of the latest code: http://github.com/joblib/joblib/archives/master
 Installing
 =========================
 
-As any Python packages, to install joblib, simply do::
+As most of pure Python packages, to install joblib, simply do::
```
Comment (lesteve) [inline README.rst]: I'd rather keep it plain and simple here:
```
You can use pip to install joblib:
```

File: README.rst [hunk 2]
```diff
@@ -15,11 +15,15 @@ of the latest code: http://github.com/joblib/joblib/archives/master
...
 
-    python setup.py install
+    pip install joblib
 
-in the source code directory.
+from any directory or
+
+    pip install -e .
```
Comment (lesteve) [inline README.rst]: Leave the python setup.py install. For the record `-e` is the equivalent of `python setup.py develop`.

File: doc/installing.rst [hunk 3]
```diff
@@ -1,42 +1,35 @@
...
+    pip install --prefix /usr/local joblib
 
 * Installing only for a specific user is easy if you use Python 2.6 or
   above::
 
-    easy_install --user joblib
+    pip install --user joblib
 
-.. warning::
+Using `conda`
```
Comment (lesteve) [inline doc/installing.rst]: I would not mention conda since as you say it is generally not the latest version.

Comment (aabadie) [inline doc/installing.rst]: except if one uses conda-forge.

Comment (lesteve) [inline doc/installing.rst]: and I don't think joblib is available on conda-forge.

File: doc/installing.rst [hunk 4]
```diff
@@ -54,7 +47,7 @@ the changes are local to your account and easy to clean up.
 Simply move to the directory created by expanding the `joblib` tarball
 and run the following command::
 
-    python setup.py install --user
+    pip install -e . --user
```
Comment (lesteve) [inline doc/installing.rst]: Remove -e just to keep it consistent to previous instructions.

File: doc/installing.rst [hunk 5]
```diff
@@ -63,9 +56,9 @@ If you have administrator rights and want to install for all users, all
 you need to do is to go in directory created by expanding the `joblib`
 tarball and run the following line::
 
-    python setup.py install
+    pip install -e .
```
Comment (lesteve) [inline doc/installing.rst]: Idem

File: doc/installing.rst [hunk 6]
```diff
@@ -63,9 +56,9 @@ If you have administrator rights and want to install for all users, all
...
 
 If you are under Unix, we suggest that you install in '/usr/local' in
 order not to interfere with your system::
 
-    python setup.py install --prefix /usr/local
+    pip install -e --prefix /usr/local
```
Comment (lesteve) [inline doc/installing.rst]: Idem

File: doc/installing.rst (hunk 3 above)
Comment (aabadie) [inline doc/installing.rst]: indeed, I got confused with what was installed and the results of `conda search` ... Let's only keep pip then.

**Comment by lesteve** [comment]
Could you update doc/installing.rst while you are at it. It does mention easy_install and I guess it would be good to double-check that everything is consistent between README.md and the rst docs.

**Comment by aabadie** [comment]
> Could you update doc/installing.rst while you are at it. It does mention easy_install and I guess it would be good to double-check that everything is consistent between README.md and the rst docs.

It was done already. I updated the changes following your comments.

File: doc/installing.rst [hunk 7]
```diff
@@ -1,42 +1,26 @@
...
 ===================
 
-The `easy_install` way
------------------------
+Using `pip`
+-----------
 
-For the easiest way to install joblib you need to have `setuptools`
-installed.
+You can use `pip` to install joblib:::
```
Comment (lesteve) [inline doc/installing.rst]: 2 `:` is enough, right?

Comment (aabadie) [inline doc/installing.rst]: Arf, fixed

File: doc/installing.rst [hunk 8]
```diff
@@ -1,42 +1,26 @@
...
-    easy_install --user joblib
-
-.. warning::
-
-    Packages installed via `easy_install` override the Python module look
-    up mechanism and thus can confused people not familiar with
-    setuptools. Although it may seem harder, we suggest that you use the
-    manual way, as described in the following paragraph.
-
-Using distributions
```
Comment (lesteve) [inline doc/installing.rst]: Why did you remove this. This is actually recommended if you care about stability and ease of maintenance.

File: README.rst [hunk 9]
```diff
@@ -15,11 +15,15 @@ of the latest code: http://github.com/joblib/joblib/archives/master
...
-As any Python packages, to install joblib, simply do::
+As most of pure Python packages, to install joblib, simply do::
 
-    python setup.py install
+    pip install joblib
 
-in the source code directory.
+from any directory or
+
+    pip install .
```
Comment (lesteve) [inline README.rst]: I would keep python setup.py install. Not sure what `pip install .` brings over `python setup.py install`. I think pip actually looks at setup.py and uses the info there.

File: doc/installing.rst (hunk 8 above)
Comment (aabadie) [inline doc/installing.rst]: Initially, I thought that using `pip` was the best/recommended choice but I agree I should revert this.

File: doc/installing.rst [hunk 10]
```diff
@@ -54,7 +38,7 @@ the changes are local to your account and easy to clean up.
 Simply move to the directory created by expanding the `joblib` tarball
 and run the following command::
 
-    python setup.py install --user
+    pip install . --user
```
Comment (aabadie) [inline doc/installing.rst]: I guess I should revert this and the lines below as well ?

**Comment by aabadie** [comment]
@lesteve, I addressed your previous comments.

**Comment by lesteve** [comment]
There are still 2 unaddressed comments. I'll merge this one once they are tackled.

**Comment by aabadie** [comment]
> There are still 2 unaddressed comments

Should be ok now. I can squash the commits before merging if it's fine now.

**Comment by lesteve** [comment]
LGTM merging.

> I can squash the commits before merging if it's fine now.

No real need you can squash and merge through the Github interface these days.
